File: android/api-tests/src/test/java/com/revenuecat/apitests/java/HybridPurchaseLogicBridgeApiTests.java
package com.revenuecat.apitests.java;

import com.revenuecat.purchases.hybridcommon.ui.HybridPurchaseLogicBridge;
import com.revenuecat.purchases.ui.revenuecatui.PurchaseLogic;

import kotlin.Unit;

@SuppressWarnings({"unused"})
class HybridPurchaseLogicBridgeApiTests {

    private void checkConstants() {
        String requestIdKey = HybridPurchaseLogicBridge.EVENT_KEY_REQUEST_ID;
        String packageKey = HybridPurchaseLogicBridge.EVENT_KEY_PACKAGE_BEING_PURCHASED;

        String success = HybridPurchaseLogicBridge.RESULT_SUCCESS;
        String cancellation = HybridPurchaseLogicBridge.RESULT_CANCELLATION;
        String error = HybridPurchaseLogicBridge.RESULT_ERROR;
    }

    private void checkConstructor() {
        HybridPurchaseLogicBridge bridge = new HybridPurchaseLogicBridge(
                eventData -> { return Unit.INSTANCE; },
                eventData -> { return Unit.INSTANCE; }
        );

        HybridPurchaseLogicBridge bridgeWithNulls = new HybridPurchaseLogicBridge(null, null);
    }

    private void checkImplementsPurchaseLogic() {
        HybridPurchaseLogicBridge bridge = new HybridPurchaseLogicBridge(null, null);
        PurchaseLogic purchaseLogic = bridge;
    }

    private void checkResolveResult() {
        HybridPurchaseLogicBridge.resolveResult("requestId", HybridPurchaseLogicBridge.RESULT_SUCCESS, null);
        HybridPurchaseLogicBridge.resolveResult("requestId", HybridPurchaseLogicBridge.RESULT_ERROR, "error message");
    }

    private void checkCancelPending() {
        HybridPurchaseLogicBridge bridge = new HybridPurchaseLogicBridge(null, null);
        bridge.cancelPending();
    }
}

File: android/api-tests/src/test/java/com/revenuecat/apitests/kotlin/HybridPurchaseLogicBridgeApiTests.kt
package com.revenuecat.apitests.kotlin

import com.revenuecat.purchases.hybridcommon.ui.HybridPurchaseLogicBridge
import com.revenuecat.purchases.ui.revenuecatui.PurchaseLogic

@Suppress("unused", "UNUSED_VARIABLE", "EmptyFunctionBlock")
private class HybridPurchaseLogicBridgeApiTests {

    fun checkConstants() {
        val requestIdKey: String = HybridPurchaseLogicBridge.EVENT_KEY_REQUEST_ID
        val packageKey: String = HybridPurchaseLogicBridge.EVENT_KEY_PACKAGE_BEING_PURCHASED

        val success: String = HybridPurchaseLogicBridge.RESULT_SUCCESS
        val cancellation: String = HybridPurchaseLogicBridge.RESULT_CANCELLATION
        val error: String = HybridPurchaseLogicBridge.RESULT_ERROR
    }

    fun checkConstructor() {
        val bridge = HybridPurchaseLogicBridge(
            onPerformPurchase = { eventData -> },
            onPerformRestore = { eventData -> },
        )

        val bridgeWithNulls = HybridPurchaseLogicBridge(
            onPerformPurchase = null,
            onPerformRestore = null,
        )
    }

    fun checkImplementsPurchaseLogic() {
        val bridge = HybridPurchaseLogicBridge(null, null)
        val purchaseLogic: PurchaseLogic = bridge
    }

    fun checkResolveResult() {
        HybridPurchaseLogicBridge.resolveResult("requestId", HybridPurchaseLogicBridge.RESULT_SUCCESS)
        HybridPurchaseLogicBridge.resolveResult("requestId", HybridPurchaseLogicBridge.RESULT_ERROR, "error message")
    }

    fun checkCancelPending() {
        val bridge = HybridPurchaseLogicBridge(null, null)
        bridge.cancelPending()
    }
}
